services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/task.py

```diff
@@ -284,6 +287,88 @@ async def retrieve_service_inputs(
 
         return RetrieveDataOutEnveloped.from_transferred_bytes(transferred_bytes)
 
+    async def attach_sharing_network(
+        self, node_id: NodeID, network_id: str, network_alias: DockerNetworkAlias
+    ) -> None:
+        if node_id not in self._inverse_search_mapping:
+            logger.debug("No dynamic-sidecar running for %s", f"{node_id=}")
+            return
+
+        service_name = self._inverse_search_mapping[node_id]
+        scheduler_data = self._to_observe[service_name].scheduler_data
+
+        dynamic_sidecar_client: DynamicSidecarClient = get_dynamic_sidecar_client(
+            self.app
+        )
+
+        try:
+            containers_status = await dynamic_sidecar_client.containers_docker_status(
+                dynamic_sidecar_endpoint=scheduler_data.dynamic_sidecar.endpoint
+            )
+        except httpx.HTTPError:
+            logger.debug("Could not contact dynamic-sidecar for %s", f"{node_id=}")
+            return
+
+        sorted_container_names = sorted(containers_status.keys())
+
+        entrypoint_container_name = await dynamic_sidecar_client.get_entrypoint_container_name(
+            dynamic_sidecar_endpoint=scheduler_data.dynamic_sidecar.endpoint,
+            dynamic_sidecar_network_name=scheduler_data.dynamic_sidecar_network_name,
+        )
+
+        tasks = deque()
+
+        for k, container_name in enumerate(sorted_container_names):
+            # by default we attach `alias-0`, `alias-1`, etc...
+            # to all containers
+            aliases = [f"{network_alias}-{k}"]
+            if container_name == entrypoint_container_name:
+                # by definition the entrypoint container will be exposed as the `alias`
+                aliases.append(network_alias)
+
+            tasks.append(
+                dynamic_sidecar_client.attach_container_to_network(
+                    dynamic_sidecar_endpoint=scheduler_data.dynamic_sidecar.endpoint,
+                    container_id=container_name,
+                    network_id=network_id,
+                    network_aliases=aliases,
+                )
+            )
+
+        await logged_gather(*tasks)
+
+    async def detach_sharing_network(self, node_id: NodeID, network_id: str) -> None:
+        if node_id not in self._inverse_search_mapping:
+            logger.debug("No dynamic-sidecar running for %s", f"{node_id=}")
```

Comment:
does this happen? why and should it be only logged as a debug message?

Reply:
Will log as info. But yes it happens. Because the dynamic-sidecar might not be in place and the request might still arrive to this endpoint. It is fine to do nothing.

Reply:
same here